src/test/java/edu/hm/hafner/util/SetTest2.java

```diff
@@ -0,0 +1,179 @@
+        testHash.remove(471);
+        assertThat(testHash.iterator().hasNext()).isFalse();
+    }
+
+    @Test
+    void size() {
+        testInitializing();
+
+        //Test: Sizes after declaration.
+        assertThat(testHash.size()).isEqualTo(0);
+        assertThat(extendHash.size()).isEqualTo(4);
+
+        // Test: Adding values
+        testHash.addAll(extendHash);
+        assertThat(testHash.size()).isEqualTo(4);
+        testHash.add(555);
+        assertThat(testHash.size()).isEqualTo(5);
+        testHash.add(555);
+        assertThat(testHash.size()).isEqualTo(5);
+
+        // Test: Size after removing a value.
+        testHash.remove(555);
+        assertThat(testHash.size()).isEqualTo(4);
+        testHash.removeAll(extendHash);
+        assertThat(testHash.size()).isEqualTo(0);
+    }
+
+    @Test
+    void isEmpty() {
+        testInitializing();
+
+        //Test: Creating a new variable
+        assertThat(testHash.isEmpty()).isTrue();
+
+        //Test: Adding values, so the hash is not empty any more.
+        testHash.add(1997);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.add(1999);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.add(1900);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.add(123);
+        assertThat(testHash.isEmpty()).isFalse();
+
+        //Test: Removing all values in the hash step by step.
+        testHash.remove(1997);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.remove(1999);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.remove(1900);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.remove(123);
+        assertThat(testHash.isEmpty()).isTrue();
+
+        // Test: Adding and removing more than one Object.
+        testHash.addAll(extendHash);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.removeAll(extendHash);
+        assertThat(testHash.isEmpty()).isTrue();
+    }
+
+    @Test
+    void contains() {
+        testInitializing();
+        assertThat(extendHash.contains(1999)).isTrue();
+        assertThat(extendHash.contains(1997)).isTrue();
+        assertThat(extendHash.contains(1900)).isTrue();
+        assertThat(extendHash.contains(123)).isTrue();
+        assertThat(extendHash.contains(4)).isFalse();
+        assertThat(extendHash.contains(0)).isFalse();
+        assertThat(extendHash.contains(1860)).isFalse();
+    }
+
+    @Test
+    void add() {
+        testHash = new HashSet<>();
+        assertThat(testHash).isEmpty();
+        testHash.add(11);
+        assertThat(testHash.hashCode()).isEqualTo(11);
+        testHash.add(300);
+        assertThat(testHash.hashCode()).isEqualTo(311); //11+300
+    }
+
+    @Test
+    void remove() {
+        testInitializing();
+        assertThat(extendHash).isNotNull();
+        assertThat(extendHash.hashCode()).isEqualTo(6019); //All extendedHash integers summed
```

Comment:
In Tests möglichst nicht mit hashCode arbeiten. Der Test und seine Korrektheit sind so nur schwer nachzuvollziehen.